### PR TITLE
feat: add auth

### PR DESCRIPTION
**Pull request by Bishoy-at-pieces**

### CODE REVIEW

File: rplugin/python3/pieces_python/auth.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+class Auth:
+	user_profile:Optional[UserProfile] = None
+	
+	@classmethod
+	def on_user_callback(cls,user):
+		cls.user_profile = user
+
+	def login(self):
+		t = OSApi(Settings.api_client).sign_into_os(async_req=True)
```
Comment (mark-at-pieces) [inline rplugin/python3/pieces_python/auth.py:14]: why are we using 't' here in the file?

**Review by mark-at-pieces — APPROVED** [review]
everything is looking good, just not sure about the 't'

**Comment by mark-at-pieces** [comment]
and looks like some minor merge conflicts

**Comment by Bishoy-at-pieces** [comment]
> everything is looking good, just not sure about the 't'

I can rename it to thread

**Pull request merged**